### PR TITLE
bot: Build suite, chunk and platform from full task

### LINKED ISSUE #20: Get suite and chunk name without parsing the task name
The suite and chunk info are already in the task data, we don't need to parse them from the task name:
```
   {
      "status": {
        "taskId": "CkICVuBvRr--6rratJoG1A",
        "provisionerId": "aws-provisioner-v1",
        "workerType": "gecko-t-linux-large",
        "schedulerId": "gecko-level-3",
        "taskGroupId": "e4wR3JiKRmCqFXMG1sZoqw",
        "deadline": "2018-01-16T14:08:30.100Z",
        "expires": "2019-01-15T14:08:30.100Z",
        "retriesLeft": 5,
        "state": "completed",
        "runs": [
          {
            "runId": 0,
            "state": "completed",
            "reasonCreated": "scheduled",
            "reasonResolved": "completed",
            "workerGroup": "us-west-2",
            "workerId": "i-02f880a62021594ec",
            "takenUntil": "2018-01-15T15:18:41.003Z",
            "scheduled": "2018-01-15T14:58:40.472Z",
            "started": "2018-01-15T14:58:41.080Z",
            "resolved": "2018-01-15T15:04:13.243Z"
          }
        ]
      },
      "task": {
        "provisionerId": "aws-provisioner-v1",
        "workerType": "gecko-t-linux-large",
        "schedulerId": "gecko-level-3",
        "taskGroupId": "e4wR3JiKRmCqFXMG1sZoqw",
        "dependencies": [
          "HyAoNcNPRvyV6mgzFFFWSg",
          "XUDNfri0ReazT-7gyTTAYg"
        ],
        "requires": "all-completed",
        "routes": [
          "tc-treeherder.v2.mozilla-central.a00e3a3dbad43759ee9474d6c7c908fa16deaee8.33255",
          "coalesce.v1.mozilla-central.71fe57dadaef770d25ba"
        ],
        "priority": "medium",
        "retries": 5,
        "created": "2018-01-15T14:08:30.100Z",
        "deadline": "2018-01-16T14:08:30.100Z",
        "expires": "2019-01-15T14:08:30.100Z",
        "scopes": [
          "secrets:get:project/taskcluster/gecko/hgfingerprint",
          "docker-worker:feature:allowPtrace",
          "docker-worker:cache:level-3-mozilla-central-test-workspace-bc7e1a7ad01a345394f1",
          "docker-worker:cache:level-3-checkouts-bc7e1a7ad01a345394f1"
        ],
        "payload": {
          "supersederUrl": "https://coalesce.mozilla-releng.net/v1/list/3600/5/mozilla-central.71fe57dadaef770d25ba",
          "onExitStatus": {
            "retry": [
              4
            ]
          },
          "maxRunTime": 3600,
          "image": {
            "path": "public/image.tar.zst",
            "type": "task-image",
            "taskId": "XUDNfri0ReazT-7gyTTAYg"
          },
          "cache": {
            "level-3-mozilla-central-test-workspace-bc7e1a7ad01a345394f1": "/builds/worker/workspace",
            "level-3-checkouts-bc7e1a7ad01a345394f1": "/builds/worker/checkouts"
          },
          "artifacts": {
            "public/logs/": {
              "path": "/builds/worker/workspace/build/upload/logs/",
              "expires": "2019-01-15T14:08:30.100Z",
              "type": "directory"
            },
            "public/test": {
              "path": "/builds/worker/artifacts/",
              "expires": "2019-01-15T14:08:30.100Z",
              "type": "directory"
            },
            "public/test_info/": {
              "path": "/builds/worker/workspace/build/blobber_upload_dir/",
              "expires": "2019-01-15T14:08:30.100Z",
              "type": "directory"
            }
          },
          "command": [
            "/builds/worker/bin/run-task",
            "--",
            "/builds/worker/bin/test-linux.sh",
            "--no-read-buildbot-config",
            "--installer-url=https://queue.taskcluster.net/v1/task/HyAoNcNPRvyV6mgzFFFWSg/artifacts/public/build/target.tar.bz2",
            "--test-packages-url=https://queue.taskcluster.net/v1/task/HyAoNcNPRvyV6mgzFFFWSg/artifacts/public/build/target.test_packages.json",
            "--reftest-suite=reftest-no-accel",
            "--e10s",
            "--allow-software-gl-layers",
            "--total-chunk=8",
            "--this-chunk=7",
            "--download-symbols=ondemand"
          ],
          "env": {
            "SCCACHE_DISABLE": "1",
            "MOZ_NODE_PATH": "/usr/local/bin/node",
            "TASKCLUSTER_CACHES": "/builds/worker/checkouts;/builds/worker/workspace",
            "HG_STORE_PATH": "/builds/worker/checkouts/hg-store",
            "GECKO_HEAD_REV": "a00e3a3dbad43759ee9474d6c7c908fa16deaee8",
            "GECKO_HEAD_REPOSITORY": "https://hg.mozilla.org/mozilla-central",
            "TASKCLUSTER_VOLUMES": "/builds/worker/.cache;/builds/worker/checkouts;/builds/worker/tooltool-cache;/builds/worker/workspace",
            "MOZHARNESS_URL": "https://queue.taskcluster.net/v1/task/HyAoNcNPRvyV6mgzFFFWSg/artifacts/public/build/mozharness.zip",
            "NEED_PULSEAUDIO": "true",
            "MOZ_AUTOMATION": "1",
            "NEED_WINDOW_MANAGER": "true",
            "MOZHARNESS_CONFIG": "unittests/linux_unittest.py remove_executables.py",
            "ENABLE_E10S": "true",
            "MOZHARNESS_SCRIPT": "desktop_unittest.py",
            "GECKO_BASE_REPOSITORY": "https://hg.mozilla.org/mozilla-unified",
            "MOZILLA_BUILD_URL": "https://queue.taskcluster.net/v1/task/HyAoNcNPRvyV6mgzFFFWSg/artifacts/public/build/target.tar.bz2"
          },
          "features": {
            "taskclusterProxy": true,
            "allowPtrace": true
          }
        },
        "metadata": {
          "owner": "ncsoregi@mozilla.com",
          "source": "https://hg.mozilla.org/mozilla-central/file/a00e3a3dbad43759ee9474d6c7c908fa16deaee8/taskcluster/ci/test",
          "description": "Reftest not accelerated run ([Treeherder push](https://treeherder.mozilla.org/#/jobs?repo=mozilla-central&revision=a00e3a3dbad43759ee9474d6c7c908fa16deaee8))",
          "name": "test-linux32/opt-reftest-no-accel-e10s-7"
        },
        "tags": {
          "kind": "test",
          "worker-implementation": "docker-worker",
          "createdForUser": "ncsoregi@mozilla.com",
          "label": "test-linux32/opt-reftest-no-accel-e10s-7",
          "test-type": "reftest",
          "os": "linux"
        },
        "extra": {
          "chunks": {
            "current": 7,
            "total": 8
          },
          "suite": {
            "flavor": "reftest-no-accel",
            "name": "reftest"
          },
          "treeherder": {
            "jobKind": "test",
            "groupSymbol": "tc-R-e10s",
            "collection": {
              "opt": true
            },
            "machine": {
              "platform": "linux32"
            },
            "groupName": "Reftests executed by TaskCluster with e10s",
            "tier": 1,
            "symbol": "Ru7"
          },
          "parent": "e4wR3JiKRmCqFXMG1sZoqw",
          "index": {
            "rank": 1516025190
          }
        }
      }
    },
```

### PR DESCRIPTION
**Pull request by La0**
Fixes #20

### CODE REVIEW

**Comment by La0** [comment]
I had to keep 2 string conversions for chunk_mapping, as we do not have the full tasks there.

File: bot/code_coverage_bot/taskcluster.py [hunk 1]
```diff
@@ -124,10 +123,16 @@ def perform_download():
 
 
 def is_coverage_task(task):
-    return any(task["task"]["metadata"]["name"].startswith(t) for t in TEST_PLATFORMS)
+    return any(task["metadata"]["name"].startswith(t) for t in TEST_PLATFORMS)
+
 
+def name_to_chunk(name):
+    """
+    Helper to convert a task naem to a chunk
```
Comment (marco-c) [inline bot/code_coverage_bot/taskcluster.py]:
```suggestion
    Helper to convert a task name to a chunk
```

File: bot/tests/test_artifacts.py [hunk 2]
```diff
@@ -102,12 +102,12 @@ def test_download(
     assert mocked_get_task_artifact.call_count == 1
     assert mocked_download_artifact.call_count == 2
     assert mocked_download_artifact.call_args_list[0] == mock.call(
-        "ccov-artifacts/linux_mochitest-devtools-chrome-4_code-coverage-grcov.zip",
+        "ccov-artifacts/linux_mochitest-4_code-coverage-grcov.zip",
```
Comment (marco-c) [inline bot/tests/test_artifacts.py]: Why did this change?

Comment (La0) [reply bot/tests/test_artifacts.py]: Here we use a [fixture describing a task ](https://github.com/mozilla/code-coverage/blob/master/bot/tests/fixtures/test_task_from_group.json#L126), with `mochitest` as `suite.name`.
So the generated local filename does not have the fully extended name anymore.

Comment (La0) [reply bot/tests/test_artifacts.py]: As discussed on IRC, i reverted the unit tests by updating the fixture's suite
in a64e0a to `mochitest-devtools-chrome`

**Pull request merged**